src/osd/PeeringState.h

```diff
@@ -1765,6 +1765,17 @@ class PeeringState : public MissingLoc::MappingInfo {
     std::optional<eversion_t> trim_to,
     std::optional<eversion_t> roll_forward_to);
 
+  void append_log(
+    const vector<pg_log_entry_t>& logv,
+    eversion_t roll_forward_to,
+    ObjectStore::Transaction &t,
+    bool transaction_applied,
+    bool async) {
+    update_trim_to();
```

Comment:
I think you've moved this here to avoid having to call it where append_log is called.  However, this should only be called on the primary, whereas append_log is called on both.  The replica gets its trim_to value from the MOSDOp.

Reply:
Ok:-)